app/helpers/Intersect.js

```diff
@@ -571,3 +572,41 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+async function getPageData(authority, page) {
+  // this function should always return an object.
+  // if there's no data, return an empty object.
+  const cacheKey = authority.internal_id + '|page:' + page.id;
+  let pageData = await GetStoreData(cacheKey);
+  if (!pageData) {
+    if (await shouldDownloadPageData()) {
+      try {
```

Comment:
To clarify, are you saying for skipped pages, we manually throw an error to force the code path into the `catch` block?

I didn't intend for this function to always return `{}` on error, I meant for this to always return an object in all cases, so that the consumer of this function doesn't have to check for null.

Reply:
Yeah, similarly to what @aledustet is recommending, it reads to me like all of the `skipped` pages occur from errors, so it would be nice for logging. That said, logging is hypothetical around this right now, so maybe just worth revisiting when introduced.